synthtool/gcp/templates/python_mono_repo_library/noxfile.py.j2

```diff
@@ -232,14 +232,27 @@ def install_unittest_dependencies(session, *constraints):
         session.install("-e", ".", *constraints)
 
 
-def default(session):
+@nox.session(python=UNIT_TEST_PYTHON_VERSIONS)
+@nox.parametrize(
+    "protobuf_implementation",
+    [ "python", "upb", "cpp" ],
+)
+def unit(session, protobuf_implementation):
     # Install all test dependencies, then install this package in-place.
 
+    if protobuf_implementation == "cpp" and session.python in ("3.11", "3.12"):
+        session.skip("cpp implementation is not supported in python 3.11+")
+
     constraints_path = str(
         CURRENT_DIRECTORY / "testing" / f"constraints-{session.python}.txt"
     )
     install_unittest_dependencies(session, "-c", constraints_path)
 
+    # Remove the 'cpp' implementation once support for Protobuf 3.x is dropped.
```

Comment:
nit: add `TODO`

Reply:
Done in https://github.com/googleapis/synthtool/pull/1977